scripts/generate.py

```diff
@@ -0,0 +1,381 @@
+    out_file = os.path.join(RUN, f"kraft-{suffix}-{run['name']}.sh")
+    header = TEMPLATE_RUN_KRAFT_HEADER.format(run["command"])
+    with open(out_file, "w", encoding="utf8") as stream:
+        stream.write(header)
+        if "rootfs" in run.keys():
+            stream.write("rootfs=\"{run['rootfs']}\"\n")
+        stream.write(RUN_KILL_COMMANDS)
+        if run["networking"]:
+            stream.write(RUN_COMMON_NET_COMMANDS)
+            stream.write(RUN_KRAFT_NET_COMMANDS)
+        stream.write("\n")
+        if run["networking"]:
+            stream.write("sudo ")
+        stream.write("kraft run \\\n")
+        stream.write("    --log-level debug --log-type basic \\\n")
+        if run["networking"]:
+            stream.write("    --network bridge:virbr0 \\\n")
+        stream.write(f"    --target {target} --plat {plat} \\\n")
+        if "rootfs" in run.keys():
+            if fs == "initrd":
+                stream.write("    --initrd \"$rootfs\" \\\n")
+            if fs == "9pfs":
+                stream.write("    -v \"$rootfs\":/ \\\n")
+        if run["networking"]:
+            stream.write("    -a netdev.ipv4_addr=172.44.0.2 -a netdev.ipv4_gw_addr=172.44.0.1 \
+-a netdev.ipv4_subnet_mask=255.255.255.0 \\\n")
+        stream.write("    -- $cmd \\\n")
+
+        st = os.stat(out_file)
+        os.chmod(out_file, st.st_mode | stat.S_IEXEC)
+
+def generate_run():
+    """Generate running scripts.
+
+    Scripts are generated in scripts/run/ directory.
+    """
+
+    if not os.path.exists(RUN):
+        os.mkdir(RUN)
+
+    # Obtain configurations for running applications.
+    with open(os.path.join(SCRIPTS, "run.yaml"), "r", encoding="utf8") as stream:
+        data = yaml.safe_load(stream)
+        runs = data['runs']
+
+    # Obtain targets for basic QEMU and Firecracker-based runs from defconfigs.
+    def_targets = []
+    for file in files(DEFCONFIGS):
+        def_targets.append(tuple(file.split('-')[0:3]))
+    def_targets = set(def_targets)
+
+    # Obtain targets for KraftKit runs form kraft.yaml.
+    with open(KRAFTCONFIG, "r", encoding="utf8") as stream:
+        data = yaml.safe_load(stream)
+        kraft_targets = data['targets']
+
+    # Walk through all configured applications.
+    for run in runs:
+        # Generate QEMU/Firecrakcer based scripts based on defconfigs.
+        for plat, arch, fs in def_targets:
+            if plat == "fc":
+                generate_run_fc(run, arch, fs)
+            if plat == "qemu":
+                generate_run_qemu(run, arch, fs)
+
+        # Generate KraftKit-based run scripts from kraft.yaml targets.
+        for target in kraft_targets:
+            fs = None
+            if "CONFIG_LIBVFSCORE_ROOTFS=\"9pfs\"" in target['kconfig']:
+                fs = "9pfs"
+            elif "CONFIG_LIBVFSCORE_ROOTFS=\"initrd\"" in target['kconfig']:
+                fs = "initrd"
+            generate_run_kraft(run, target['name'], target['platform'], fs)
```

Comment:
```suggestion
        for target in kraft_targets:
            fs = None
            if 'kconfig' not in target:
                fs = ""
            elif "CONFIG_LIBVFSCORE_ROOTFS=\"9pfs\"" in target['kconfig']:
                fs = "9pfs"
            elif "CONFIG_LIBVFSCORE_ROOTFS=\"initrd\"" in target['kconfig']:
                fs = "initrd"
            generate_run_kraft(run, target['name'], target['platform'], fs)
```
This will fail if there is no kconfig list in kraft.yaml